test/e2e/antreapolicy_test.go

```diff
@@ -3000,6 +3001,246 @@ func testACNPICMPSupport(t *testing.T, data *TestData) {
+		receiverConfigs: []multicastTestPodConfig{{1, false}},
+		port:            3457,
+		group:           net.ParseIP("224.3.4.16"),
+	}
+	testACNPMulticastEgress(t, data, name, mc, nodeMulticastInterfaces, crdv1alpha1.RuleActionDrop)
+}
+
+func testACNPMulticastEgress(t *testing.T, data *TestData, name string, mc multicastTestcase, nodeMulticastInterfaces map[int][]string, action crdv1alpha1.RuleAction) {
+	mcjoinWaitTimeout := defaultTimeout / time.Second
+	testNamespace := data.testNamespace
+	senderName, _, cleanupFunc := createAndWaitForPod(t, data, data.createMcJoinPodOnNode, "test-sender-", nodeName(mc.senderConfig.nodeIdx), testNamespace, mc.senderConfig.isHostNetwork)
+	defer cleanupFunc()
+	var wg sync.WaitGroup
+	receiverNames, cleanupFuncs := setupReceivers(t, data, mc, mcjoinWaitTimeout, &wg)
+	for _, cleanupFunc := range cleanupFuncs {
+		defer cleanupFunc()
+	}
+
+	// Wait 2 seconds(-w 2) before sending multicast traffic.
+	// It sends two multicast packets for every second(-f 500 means it takes 500 milliseconds for sending one packet).
+	sendMulticastCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 3 -w 2 -W %d %s", mc.port, mcjoinWaitTimeout, mc.group.String())}
+	go func() {
+		data.RunCommandFromPod(testNamespace, senderName, mcjoinContainerName, sendMulticastCommand)
+	}()
+	// check if receiver can receive multicast packet
+	tcpdumpName, _, cleanupFunc := createAndWaitForPod(t, data, data.createNetshootPodOnNode, "test-tcpdump-", nodeName(mc.receiverConfigs[0].nodeIdx), testNamespace, true)
+	defer cleanupFunc()
+	cmd, err := generatePacketCaptureCmd(data, 5, mc.group.String(), nodeName(mc.receiverConfigs[0].nodeIdx), receiverNames[0])
+	if err != nil {
+		t.Fatalf("failed to call generateConnCheckCmd: %v", err)
+	}
+
+	if err := wait.Poll(3*time.Second, defaultTimeout, func() (bool, error) {
+		captured, _, _, err := checkPacketCaptureResult(data, tcpdumpName, cmd)
+		if captured {
+			return true, nil
+		}
+		return false, err
+	}); err != nil {
+		t.Fatalf("Error waiting for multicast routes and stats: %v", err)
+	}
+	wg.Wait()
+	label := "multicast-egress"
+	acnpName := name
+	_, err = k8sUtils.LabelPod(testNamespace, senderName, "antrea-e2e", label)
+	if err != nil {
+		t.Fatalf("failed to label pod %s: err=%v", senderName, err)
+	}
+	// create acnp with egress rule for multicast traffic
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName(acnpName).SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{PodSelector: map[string]string{"antrea-e2e": label}}})
+	cidr := mc.group.String() + "/32"
+	builder.AddEgress(ProtocolUDP, nil, nil, nil, nil, nil, nil, nil, &cidr, nil, nil,
+		nil, nil, false, nil, action, "", "", nil)
+	acnp := builder.Get()
+	_, err = k8sUtils.CreateOrUpdateACNP(acnp)
+	defer data.crdClient.CrdV1alpha1().ClusterNetworkPolicies().Delete(context.TODO(), acnp.Name, metav1.DeleteOptions{})
```

Comment:
This defer should be placed under line 3194? If the creation is failed, no need to delete the resource?

Reply:
done